File: code_challenge_solution/python/141 Palindrome checker.py
```python
def is_palindrome(string):
    string = string.lower()
    reversed_string = string[::-1]
    return string == reversed_string

input_string = input("Enter a string: ")

if is_palindrome(input_string):
    print("The string is a palindrome.")
else:
    print("The string is not a palindrome.")
```

File: code_challenge_solution/c++/141 Palindrome checker.cpp
#include <iostream>
#include <string>
#include <algorithm>

bool isPalindrome(const std::string& str) {
    std::string reverseStr = str;
    std::reverse(reverseStr.begin(), reverseStr.end());
    return str == reverseStr;
}

int main() {
    std::string input;
    std::cout << "Enter a string: ";
    std::cin >> input;
    
    if (isPalindrome(input)) {
        std::cout << "Palindrome" << std::endl;
    } else {
        std::cout << "Not a palindrome" << std::endl;
    }
    
    return 0;
}